src/foam/graph/GraphNode.js

```diff
@@ -23,7 +23,13 @@ foam.CLASS({
     },
     {
       name: 'forwardLinks',
-      class: 'StringArray'
+      class: 'StringArray',
+      preSet: function (_, n) {
```

Comment:
Use 'adapt' instead of preSet when adapting values. You also have the foam.Array.unique() method which removes duplicate values from arrays. However, it also sorts the values.